lib/Client.js

```diff
@@ -2048,6 +2048,23 @@ class Client extends EventEmitter {
         return this.requestHandler.request("GET", Endpoints.USER(userID), true).then((user) => new User(user, this));
     }
 
+    /**
+    * Search for guild members by partial nickname/username
+    * @arg {String} guildID The ID of the guild
+    * @arg {String} query The query string to match username(s) and nickname(s) against
+    * @arg {Number} [limit=1] The maximum number of members you want returned, capped at 100
+    * @returns {Promise<Member[]>}
+    */
+    searchGuildMembers(guildID, query, limit) {
+        return this.requestHandler.request("GET", Endpoints.GUILD_MEMBERS_SEARCH(guildID), true, {
+            query,
+            limit
+        }).then((members) => {
+            const guild = this.guilds.get(guildID);
+            return members.map((member) => new Member(member, guild, this));
```

Comment:
I would really like to see if we could add these members to cache. With the changes to intents, making sure our members cache is incredibly useful/important now. 

```ts
return members.map(member => guild.members.add(member, guild, this))
```

This way it will atleast add any members not currently cached

Reply:
No.

If the library doesn't modify the cache but you want it to, you can do this yourself with the members that are returned.

If the library does modify the cache but you don't want it to, that's much harder to reverse.